src/main/java/ch/njol/skript/expressions/ExprPlayerStatistics.java

```diff
@@ -0,0 +1,200 @@
+		int pattern = matchedPattern & 1;
+
+		//noinspection unchecked
+		statistic = (Expression<Statistic>) exprs[pattern];
+		//noinspection unchecked
+		player = (Expression<OfflinePlayer>) exprs[pattern == 0 ? 2 : 0];
+		ofType = exprs[1 + pattern];
+
+		node = getNode();
+
+		return true;
+	}
+
+	@Override
+	protected Integer @Nullable [] get(Event event) {
+		Statistic statistic = this.statistic.getSingle(event);
+		if (statistic == null)
+			return null;
+
+		Object ofType = this.ofType == null ? null : this.ofType.getSingle(event);
+
+		if (shouldStop(statistic, ofType))
+			return null;
+
+		return player.stream(event)
+			.map(player -> getStatistic(player, statistic, ofType))
+			.toArray(Integer[]::new);
+	}
+
+	@Override
+	public Class<?> @Nullable [] acceptChange(ChangeMode mode) {
+		return switch (mode) {
+			case SET, DELETE, RESET, REMOVE, ADD -> CollectionUtils.array(Integer.class);
+			default -> null;
+		};
+	}
+
+	@Override
+	public void change(Event event, Object @Nullable [] delta, ChangeMode mode) {
+		Statistic statistic = this.statistic.getSingle(event);
+		if (statistic == null)
+			return;
+
+		Object ofType = this.ofType == null ? null : this.ofType.getSingle(event);
+
+		if (shouldStop(statistic, ofType))
+			return;
+
+		int value = delta == null ? 0 : (Integer) delta[0];
+
+		for (OfflinePlayer player : this.player.getArray(event)) {
+			switch (mode) {
+				case SET -> applyStatistic(player, statistic, value, ofType);
+				case ADD -> applyStatistic(player, statistic, getStatistic(player, statistic, ofType) + value, ofType);
+				case REMOVE -> applyStatistic(player, statistic, getStatistic(player, statistic, ofType) - value, ofType);
+				case DELETE, RESET -> applyStatistic(player, statistic, 0, ofType);
+			}
+		}
+	}
+
+	private int getStatistic(OfflinePlayer player, Statistic statistic, Object type) {
+		Type statisticType = statistic.getType();
+		if (type instanceof ItemType item && (statisticType == Type.ITEM || statisticType == Type.BLOCK)) {
+			return player.getStatistic(statistic, item.getMaterial());
+		} else if (type instanceof EntityData<?> data && statisticType == Type.ENTITY) {
+			return player.getStatistic(statistic, EntityUtils.toBukkitEntityType(data));
+		}
+
+		return player.getStatistic(statistic);
+	}
+
+	private void applyStatistic(OfflinePlayer player, Statistic statistic, Integer value, Object type) {
+		if (value < 0)
+			value = 0;
+
+		Type statisticType = statistic.getType();
+
+		if (type instanceof ItemType item && (statisticType == Type.ITEM || statisticType == Type.BLOCK)) {
+			player.setStatistic(statistic, item.getMaterial(), value);
+		} else if (type instanceof EntityData<?> data && statisticType == Type.ENTITY) {
+			player.setStatistic(statistic, EntityUtils.toBukkitEntityType(data), value);
+		} else {
+			player.setStatistic(statistic, value);
+		}
+	}
+
+	private boolean shouldStop(Statistic statistic, Object ofType) {
```

Comment:
```suggestion
	private boolean checkTyping(Statistic statistic, Object ofType) {
```
some short javadocs for this method and the others would be nice